Core/clim-core/gadgets/menu.lisp

```diff
@@ -196,13 +204,10 @@
          (type (command-menu-item-type item))
          (value (command-menu-item-value item))
          (text-style (command-menu-item-text-style item))
-         (frame (pane-frame client))
-         (manager (frame-manager frame)))
+         (frame *application-frame*))
     (flet ((make-sub-pane (class &rest initargs &key &allow-other-keys)
-             (apply #'make-pane-1 manager frame class
+             (apply #'make-pane class
```

Comment:
Why this change? I have seen many problems caused by the combination of nested command loops and `*application-frame*`. Using the "hard-wired" frame instead of the special variable seems more robust.

Reply:
In fact, in relation to the other changes: the client of a gadget is usually the frame. Maybe that logic would make sense here as well.

Reply:
it is not usually the frame. it is the frame only for top-level entries, for subtrees it is the rack pane. I've changed it to use typecase (with a fallback otherwise clause to *application-frame*).